internal/cloud/realm/app.go

```diff
@@ -156,3 +195,13 @@ func (c *client) getApps(groupID string) ([]App, error) {
 	}
 	return apps, nil
 }
+
+func appsPathWithProduct(groupID, product string) string {
```

Comment:
can this just be a simple if condition to check if `product != ""`? do we need this as an abstracted func?

Reply:
i put it in a func for some quick unit tests, there's no real straightforward way currently for the CLI to assert on request URLs sent from the realm client (this probably _should_ exist...)

i can remove the func and tests if you feel strongly against it, its relatively simple/straightforward logic.  long-term i think I'd rather put effort into adding an easy way to test the request urls we attempt to hit when sending client requests, maybe as a new-eng/quick win ticket?

Reply:
Honestly, I would just remove this - the test doesn't benefit it us much from my perspective. I would opt for a TODO on the longterm approach here instead.

Reply:
👍 https://jira.mongodb.org/browse/REALMC-8886